src/pcap/cmd/pcap-bosh-cli/main.go

```diff
@@ -0,0 +1,285 @@
+	if err != nil {
+		return
+	}
+	err = environment.UpdateTokens()
+	if err != nil {
+		return
+	}
+	err = writeBoshConfig(boshConfig, opts.BoshConfigFilename)
+	if err != nil {
+		return
+	}
+
+	logger.Debug("bosh-config and tokens successfully updated")
+
+	// set up pcap-client/pcap-api connection
+	client, err := pcap.NewClient(opts.File)
+	if err != nil {
+		err = fmt.Errorf("could not set up pcap-client %w", err)
+		return
+	}
+	err = client.ConnectToAPI(apiURL)
+	if err != nil {
+		err = fmt.Errorf("could not connect to pcap-api %w", err)
+		return
+	}
+
+	logger.Debug("pcap-client successfully initialized and connected to pcap-api")
+
+	// set up capture request
+	ctx := context.Background()
+	ctx, cancel := pcap.WithCancelCause(ctx)
+	setupContextCancel(cancel)
+	endpointRequest := createEndpointRequest(environment.AccessToken, opts.Deployment, opts.InstanceGroups)
+	captureOptions := createCaptureOptions(opts.Interface, opts.Filter, 65_000) // TODO: get snaplen from config or parameters
+
+	// perform capture request
+	err = client.HandleRequest(endpointRequest, captureOptions, ctx, cancel)
+	if err != nil {
+		err = fmt.Errorf("encountered error during request handling: %w", err)
+		return
+	}
+
+	// handle results of capture request
+	err = pcap.Cause(ctx)
+	if err != nil {
+		if status.Code(err) == codes.OK {
+			logger.Info("capture finished successfully")
+			return
+		}
+		err = fmt.Errorf("finished with error %w", err)
+		return
+	}
+}
+
+// sets up the zap.Logger. Currently outputs to stderr in Console format.
+func setupLogging() {
+	atomicLogLevel = zap.NewAtomicLevelAt(zap.WarnLevel)
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	outputPaths := []string{"stderr"} //TODO: make configurable?
+	zapConfig := zap.Config{
+		Level:             atomicLogLevel,
+		DisableCaller:     true,
+		DisableStacktrace: true,
+		Encoding:          "console",
+		EncoderConfig:     encoderConfig,
+		OutputPaths:       outputPaths,
+		ErrorOutputPaths:  outputPaths,
+	}
+	logger = zap.Must(zapConfig.Build())
+	zap.ReplaceGlobals(logger)
+	logger.Debug("successfully set up logger", zap.Strings("output-paths", outputPaths))
+}
+
+// setLogLevel sets the log level of the zap.Logger created in setupLogging via atomicLogLevel
+// It accepts []bool as parameter as this is the output of the opts.Verbose flag:
+// -v sets the level to zapcore.InfoLevel, -vv (and more v's) sets zapcore.DebugLevel.
+func setLogLevel(verbose []bool) {
+	var logLevel zapcore.Level
+	switch len(verbose) {
+	case 0:
+		logLevel = zapcore.WarnLevel
+	case 1:
+		logLevel = zapcore.InfoLevel
+	default: // if more than one -v is given as argument
+		logLevel = zapcore.DebugLevel
+	}
+
+	atomicLogLevel.SetLevel(logLevel)
+
+	logger.Debug("set log-level", zap.String("log-level", logLevel.String()))
+}
```

Comment:
This is taken from the `pcap-api`. Because this is a CLI, default could / should be INFO.